NAS-111803 / 21.08 / NAS-111803 - Allow ownership information to be changed in an ACL Editor.

undsoft opened the pull request:
Ownership information can now be updated in an acl editor.

Go to Acl Editor on datasets with NFS and POSIX entries and play around with ACL editor, especially with user and group fields.

bugclerk commented:
Jira URL: https://jira.ixsystems.com/browse/NAS-111803

RehanY147 commented on src/app/pages/storage/volumes/permissions/stores/dataset-acl-editor.store.ts:
```diff
@@ -312,54 +313,102 @@ export class DatasetAclEditorStore extends ComponentStore<DatasetAclEditorState>
+
+          return {
+            ...aceAttributes,
+            id: -1, // -1 is effectively null for middleware
```
Is there a specific reason why we're sending `-1` instead of `null` here? If not, it just looks like it'll add unnecessary confusion. If we mean to send `null`, we should just send `null`

undsoft replied:
Unfortunately `null` is not supported :)

dszidi commented:
The layout could use a little improvement.
If we kept the new fields inline it would save space a look a little tighter too.
Also I was thinking perhaps we should be using "Access Control Entry" instead of "List" on the right.
Here's a screenshot for an example.

![image](https://user-images.githubusercontent.com/29865743/129418591-5ce5a821-deae-40cd-a505-c01c1cc836c1.png)



Otherwise everything seems to work just fine in testing so far

undsoft commented:
@dszidi 
Updated text, thanks.
As for layout, while I agree that current layout is not ideal, I can't say that I like the proposed change more. It does utilize space more efficiently, but it kinda breaks the flow of the page. 
The left side is currently for things that relate to acl as a whole: buttons, checkboxes that affect saving, while the right side is for narrowing down to an individual ace. 
That's the reason I want to keep ownership dropdowns somewhere on the left, as they feel more 'global'.

I've also tried putting them in the sidebar, but that has its own problems.
<img width="1783" alt="Снимок экрана 2021-08-16 в 10 58 36" src="https://user-images.githubusercontent.com/1481270/129531475-cd26da6b-5800-48a5-abf7-1e5044c51f7f.png">

So far I've adjusted margins a little bit.

dszidi commented:
LGTM!

RehanY147 reviewed (approved):
Looks good
